optionlab/engine.py

```diff
@@ -35,31 +33,36 @@
 
 
 class StrategyEngine:
-    def __init__(self, inputs: Inputs):
+    def __init__(self, inputs_data: Inputs | dict):
```

Comment:
This is good. Python dictionaries are a more familiar data structure, I think it is easier for the users to provide input like that. The only thing is that I would prefer to provide the strategy as a separate dictionary, because the other inputs are simulation parameters.

Reply:
Moreover, it seems quite simple to get rid of this StrategyEngine (why did you append Engine to the class name?) and replace it by a run_strategy() function (is "run strategy" English? maybe eval_strategy()) that will be the only thing that the user -- no matter if beginner or advanced -- will have to call to evaluate a strategy.

Reply:
I think I'll keep it for users that already implemented it but have functions as an alternative

Reply:
> This is good. Python dictionaries are a more familiar data structure, I think it is easier for the users to provide input like that. The only thing is that I would prefer to provide the strategy as a separate dictionary, because the other inputs are calculation parameters, not input data.

Not sure. I want to keep it relatively simple. 2 different types is enough. If you accept string, it could be confusing, as users might not provide JSON-parseable and then there are errors

Reply:
> Moreover, it seems quite simple to get rid of this StrategyEngine (why did you append Engine to the class name?) and replace it by a run_strategy() function (is "run strategy" English? maybe eval_strategy()) that will be the only thing that the user -- no matter if beginner or advanced -- will have to call to evaluate a strategy.

I appended Engine to the name because we know have a type for `Strategy` in models.py